### DISCUSSION TITLE
TCP Conn Failed : error 0:61

### DISCUSSION THREAD

**Discussion by kokozakotami** [HaishinKit/HaishinKit.swift#1060] [Q&A]
Hello. I'am trying to start RTMP stream. But connection always closed and cant connect.
I've tried example and any others ways.
Sample code:
```
rtmpConnection.connect("rtmp://192.168.31.69:8080/free")
rtmpStream.publish("stream")
```
In console i can see TCP Conn Failed : error 0:61 every time when i trying to connect.
Should i start RTMP server by myself and after connect using rtmpConnection or rtmpConnection start server?

And after that i am trying to write:
```
rtmpStream.appendSampleBuffer(sampleBuffer, withType: .video)
```

**Comment by kokozakotami** [comment]
Also i've tried to test connection using VLC player but i've got an error.

**Comment by kokozakotami** [comment]
![Screenshot](https://user-images.githubusercontent.com/1745803/189491578-93f60559-32da-4864-a32c-e645b1d35735.png)
More logs

**Comment by kokozakotami** [comment]
As I understand it - I need a server. Advice on what can be used for a local server?

**Reply by shogo4405** [reply]
I recommend a server:
- nginx + rtmp module
  - https://github.com/arut/nginx-rtmp-module
- Wowza
  - https://www.wowza.com/

**Reply by kokozakotami** [reply]
I need simple iOS server (in local network). Is it possible? I've tried to use HLS but faced with 3-6 secs delay.
